doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown

```diff
@@ -0,0 +1,298 @@
+Detection of ArUco Boards {#tutorial_aruco_board_detection}
+==============================
+
+@prev_tutorial{tutorial_aruco_detection}
+@next_tutorial{tutorial_charuco_detection}
+
+An ArUco Board is a set of markers that acts like a single marker in the sense that it provides a
+single pose for the camera.
+
+The most popular board is the one with all the markers in the same plane, since it can be easily printed:
+
+![](images/gboriginal.png)
+
+However, boards are not limited to this arrangement and can represent any 2d or 3d layout.
+
+The difference between a Board and a set of independent markers is that the relative position between
+the markers in the Board is known a priori. This allows that the corners of all the markers can be used for
+estimating the pose of the camera respect to the whole Board.
+
+When you use a set of independent markers, you can estimate the pose for each marker individually,
+since you dont know the relative position of the markers in the environment.
+
+The main benefits of using Boards are:
+
+- The pose estimation is much more versatile. Only some markers are necessary to perform pose estimation.
+Thus, the pose can be calculated even in the presence of occlusions or partial views.
+- The obtained pose is usually more accurate since a higher amount of point correspondences (marker
+corners) are employed.
+
+Board Detection
+-----
+
+A Board detection is similar to the standard marker detection. The only difference is in the pose estimation step.
+In fact, to use marker boards, a standard marker detection should be done before estimating the Board pose.
+
+To perform pose estimation for boards, you should use ```#cv::solvePnP``` function, as shown below:
+
+@code{.cpp}
+cv::Mat inputImage;
+
+// Camera parameters are read from somewhere
+cv::Mat cameraMatrix, distCoeffs;
+
+// You can read camera parameters from tutorial_camera_params.yml
+readCameraParameters(filename, cameraMatrix, distCoeffs); // This function is implemented in aruco_samples_utility.hpp
+
+// Assume we have a function to create the board object
+cv::Ptr<cv::aruco::Board> board = cv::aruco::Board::create();
+
+...
+
+std::vector<int> markerIds;
+std::vector<std::vector<cv::Point2f>> markerCorners;
+
+cv::aruco::DetectorParameters detectorParams = cv::aruco::DetectorParameters();
+cv::aruco::ArucoDetector detector(board.dictionary, detectorParams);
+
+detector.detectMarkers(inputImage, markerCorners, markerIds);
+
+cv::Vec3d rvec, tvec;
+
+// If at least one marker detected
+if(markerIds.size() > 0) {
+    // Get object and image points for the solvePnP function
+    cv::Mat objPoints, imgPoints;
+    board->matchImagePoints(markerCorners, markerIds, objPoints, imgPoints);
+
+    // Find pose
+    cv::solvePnP(objPoints, imgPoints, cameraMatrix, distCoeffs, rvec, tvec);
+}
+
+@endcode
```

Comment:
Snippets should be used instead of manual **non-verified** code.

Reply:
What should I do if the example contains a lot of extra code that I would **not** like to include in the tutorial?

Reply:
Take a look on `cv::reduce` on docs.opencv.org as example.
You could define any part of the code to include into documentation.

Reply:
replaced by snippets